### PR TITLE
Add support for keyed_groups, groups, and compose options to linode i…

### PR DESCRIPTION
**Pull request by cdbunch72**
…nventory

plugin

##### SUMMARY
This adds the standard inventory plugin options keyed_groups,groups, and compose to the linode inventory plugin

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
plugins/inventory/linode.py

### CODE REVIEW

**Comment by cdbunch72** [comment]
Fixes RFE #1326

File: plugins/inventory/linode.py [hunk 1]
```diff
@@ -203,9 +219,27 @@ def parse(self, inventory, loader, path, cache=True):
 
         self._get_instances_inventory()
 
+        self._strict = self.get_option('strict')
```
Comment (felixfontein) [inline plugins/inventory/linode.py:222]: How about using a local variable here?

**Review by felixfontein — COMMENTED** [review]
Thanks for this contribution!

You need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

File: plugins/inventory/linode.py [hunk 2]
```diff
@@ -18,6 +18,8 @@
         - Uses a YAML configuration file that ends with linode.(yml|yaml).
         - Linode labels are used by default as the hostnames.
         - The inventory groups are built from groups and not tags.
+    extends_documentation_fragment:
+        - constructed
     options:
```
Comment (felixfontein) [inline plugins/inventory/linode.py:23]:
```suggestion
    options:
        strict:
            version_added: 2.0.0
        compose:
            version_added: 2.0.0
        groups:
            version_added: 2.0.0
        keyed_groups:
            version_added: 2.0.0

```
(Resp. better move them to the bottom of the `options` dict, but I cannot suggest that easily in GitHub.)

File: plugins/inventory/linode.py (hunk 1 above)
Comment (cdbunch72) [reply plugins/inventory/linode.py]: Yeah, Originally I tried to put those calls in related private methods, but the ordering did not work out, so they all got moved to parse, so an instance variable is no longer needed.  The change has been pushed.

File: plugins/inventory/linode.py (hunk 2 above)
Comment (cdbunch72) [reply plugins/inventory/linode.py:23]: Ok.  done and pushed.

File: changelogs/fragments/1453-add-support-for-keyed_groups-to-linode-inventory-plugin.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+minor_changes:
+    - community.general.linode - add the standard keyed_groups, groups, and
```
Comment (felixfontein) [inline changelogs/fragments/1453-add-support-for-keyed_groups-to-linode-inventory-plugin.yml:2]:
```suggestion
    - linode inventory plugin - add support for ``keyed_groups``, ``groups``, and
```

File: changelogs/fragments/1453-add-support-for-keyed_groups-to-linode-inventory-plugin.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+      compose options to the linode inventory plugin
```
Comment (felixfontein) [inline changelogs/fragments/1453-add-support-for-keyed_groups-to-linode-inventory-plugin.yml:3]:
```suggestion
      ``compose`` options
```

Comment (cdbunch72) [reply changelogs/fragments/1453-add-support-for-keyed_groups-to-linode-inventory-plugin.yml]: Changes implemented.

**Review by felixfontein — APPROVED** [review]
Looks reasonable. If the maintainers don't complain, I'll merge this in a week or so.

**Comment by TheLonelyGhost** [comment]
Notes on documentation:

In `linode.py` we note the dependency `linode_api4` must be `2.0.0` or higher, yet the latest version of it as of writing is `3.0.1`. Since this is a SaaS (as opposed to on-prem service) and they may start deprecating their APIs while this inventory plugin lags behind, it might be best to update the constraints there.

If that is updated, it looks like `linode_api4` only supports Python 3.6 or higher now, especially since python 2.x has been EOL'ed. Should we update these requirements, or are we certain it will work consistently with the older version of the SDK?

**Comment by cdbunch72** [comment]
I agree the python required version should probably be updated,  as anyone installing linode_api4 with pip is going to get a 3.x version (currently) and won't be able to run on 2.7. But since this would also affect the linode_v4 module, I think it needs to be a wider discussion.

File: plugins/inventory/linode.py [hunk 5]
```diff
@@ -17,7 +17,10 @@
         - Reads inventories from the Linode API v4.
         - Uses a YAML configuration file that ends with linode.(yml|yaml).
         - Linode labels are used by default as the hostnames.
-        - The inventory groups are built from groups and not tags.
+        - The default inventory groups are built from groups (deprecated by
```
Comment (felixfontein) [inline plugins/inventory/linode.py:20]:
```suggestion
        - The default inventory groups are built from groups (deprecated by
```

File: changelogs/fragments/1453-add-support-for-keyed_groups-to-linode-inventory-plugin.yml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+minor_changes:
+    - linode inventory plugin - add support for ``keyed_groups``, ``groups``,
+      and ``compose`` options 
+      (https://github.com/ansible-collections/community.general/issues/1326)
```
Comment (felixfontein) [inline changelogs/fragments/1453-add-support-for-keyed_groups-to-linode-inventory-plugin.yml:4]:
```suggestion
      (https://github.com/ansible-collections/community.general/issues/1326).
```
Sorry, I apparently overlooked that.

**Comment by felixfontein** [comment]
recheck

**Review by felixfontein — APPROVED** [review]
If nobody complains, I'll merge this in a week or so.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@cdbunch72 thanks for implementing this!
@TheLonelyGhost thanks for chiming in!

**Pull request merged**